integration-cli/docker_cli_save_load_test.go

```diff
@@ -383,3 +383,153 @@ func (s *DockerSuite) TestSaveLoadNoTag(c *check.C) {
 	c.Assert(out, checker.Contains, "Loaded image: "+name+":latest")
 	c.Assert(out, checker.Not(checker.Contains), "Loaded image ID:")
 }
+
+func (s *DockerSuite) TestSaveOCIReferencesConflicts(c *check.C) {
+	testRequires(c, DaemonIsLinux, ExperimentalDaemon)
+
+	name0 := "frombusybox0"
+	_, err := buildImage(name0, "FROM busybox\nENV oci=true", true)
+	c.Assert(err, checker.IsNil)
+
+	name1 := "frombusybox1"
+	_, err = buildImage(name1, "FROM busybox\nENV oci=true\nENV oci=false", true)
+	c.Assert(err, checker.IsNil)
+
+	// test that the same tag conflicts
+	out, _, err := dockerCmdWithError("save", "--format", "oci.v1", "-o", "test.tar", "busybox", name0)
+	c.Assert(err, checker.NotNil)
+	c.Assert(out, checker.Contains, `unable to include unique references "latest" in OCI image`)
+
+	// test that --ref on just a subset of the images raise a conflict (because name1 and busybox are still "latest")
+	out, _, err = dockerCmdWithError("save", "--format", "oci.v1", "--ref", name0+"="+name0+"-latest", "-o", "test.tar", "busybox", name0, name1)
+	c.Assert(err, checker.NotNil)
+	c.Assert(out, checker.Contains, `unable to include unique references "latest" in OCI image`)
+
+	// silly test
+	out, _, err = dockerCmdWithError("save", "--format", "oci.v1", "-o", "test.tar", "notexists")
+	c.Assert(err, checker.NotNil)
+	c.Assert(out, checker.Contains, "No such image: notexists")
+
+	// test that the same tag is invalid even if it's not "latest"
+	dockerCmd(c, "tag", name0, name0+":12.04")
+	dockerCmd(c, "tag", "busybox:latest", "busybox:12.04")
+	out, _, err = dockerCmdWithError("save", "--format", "oci.v1", "-o", "test.tar", "busybox", name0+":12.04", "busybox:12.04")
+	c.Assert(err, checker.NotNil)
+	c.Assert(out, checker.Contains, `unable to include unique references "12.04" in OCI image`)
+
+	// silly test in case you have --ref pointing to an actual tag
+	out, _, err = dockerCmdWithError("save", "--format", "oci.v1", "--ref", "busybox:latest=12.04", "-o", "test.tar", "busybox:latest", name0+":12.04")
+	c.Assert(err, checker.NotNil)
+	c.Assert(out, checker.Contains, `unable to include unique references "12.04" in OCI image`)
+
+	dockerCmd(c, "tag", "busybox:latest", "busybox0")
+	out, _, err = dockerCmdWithError("save", "--format", "oci.v1", "-o", "test.tar", "busybox:latest", "busybox0")
+	c.Assert(err, checker.NotNil)
+	c.Assert(out, checker.Contains, `unable to include unique references "latest" in OCI image`)
+
+	// test that invalid --ref aren't accepted
+	out, _, err = dockerCmdWithError("save", "--format", "oci.v1", "-o", "test.tar", "--ref", "busybox=invalid:reference", "busybox")
+	c.Assert(err, checker.NotNil)
+	c.Assert(out, checker.Contains, `invalid reference "busybox=invalid:reference", reference must not include characters outside of the set of "A" to "Z", "a" to "z", "0" to "9", the hyphen "-", the dot ".", and the underscore "_"`)
+}
+
+func (s *DockerSuite) TestSaveOCIReferences(c *check.C) {
```

Comment:
if you look at my comment above it says I need to rewrite tests because test API changed AND current integration tests are testing the old `refs` directory from image-spec. So yes, I'll do that by adapting the existing ones.